transport.go

```diff
@@ -58,37 +60,69 @@ func remapLocalHost(hostPorts []string) {
 	}
 }
 
-func parsePeer(peer string) (protocol, host string) {
-	// If we get a pure host:port, then we assume tchannel.
+// host and port must be specified in peer
+// explicitProtocol is from --protocol flag
+// protocol will be tchannel, grpc, http, https, ftp in valid case
+// boolean will be false in invalid case
+// TODO(pedge): ftp actually valid? there is a test for it
+func parsePeer(explicitProtocol string, peer string) (protocol, host string, valid bool) {
+	// If we get a pure host:port, then we assume tchannel or grpc.
```

Comment:
this whole method is very confusing after adding the `explicitProtocol` support, if there's an explicit protocol, then there's no point guessing, that should be what we always return.

basically, if `explicitProtocol` is specified, then use it. if it doesn't match with the URL protocol (e.g., `--protocol grpc --peer tchannel://foo:1234`), then return an error. otherwise, guess the protocol as we currently do.